src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1087,6 +1108,127 @@ private static FunctionResult LoadJsonContentResultBuilder(IBinder binder, IFile
+        {
+            var arguments = functionCall.Arguments.ToImmutableArray();
+            string? tokenSelectorPath = null;
+            if (arguments.Length > 1)
+            {
+                if (argumentTypes[1] is not StringLiteralType tokenSelectorType)
+                {
+                    return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[1]).CompileTimeConstantRequired()));
+                }
+                tokenSelectorPath = tokenSelectorType.RawStringValue;
+            }
+            if (!TryLoadTextContentFromFile(binder, fileResolver, diagnostics,
+                    (arguments[0], argumentTypes[0]),
+                    arguments.Length > 2 ? (arguments[2], argumentTypes[2]) : null,
+                    out var fileContent,
+                    out var errorDiagnostic,
+                    LanguageConstants.MaxJsonFileCharacterLimit))
+            {
+                return new(ErrorType.Create(errorDiagnostic));
+            }
+
+            if (ExtractTokenFromObject(fileContent) is not { } token)
+            {
+                // Instead of catching and returning the JSON parse exception, we simply return a generic error.
+                // This avoids having to deal with localization, and avoids possible confusion regarding line endings in the message.
+                return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[0]).UnparseableJsonType()));
+            }
+
+            if (tokenSelectorPath is not null)
+            {
+                try
+                {
+                    var selectTokens = token.SelectTokens(tokenSelectorPath, false).ToList();
+                    switch (selectTokens.Count)
+                    {
+                        case 0:
+                            return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[1]).NoJsonTokenOnPathOrPathInvalid()));
+                        case 1:
+                            token = selectTokens.First();
+                            break;
+                        default:
+                            token = new JArray();
+                            foreach (var selectToken in selectTokens)
+                            {
+                                ((JArray)token).Add(selectToken);
+                            }
+                            break;
+                    }
+                }
+                catch (JsonException)
+                {
+                    //path is invalid or user hasn't finished typing it yet
+                    return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[1]).NoJsonTokenOnPathOrPathInvalid()));
+                }
+            }
+
+            return new(ConvertJsonToBicepType(token), ConvertJsonToExpression(token));
+        }
+
+        [Obsolete("This method has been replaced by ExtractTokenFromObject which supports both YAML and JSON")]
```

Comment:
Why keep this around?

Reply:
we will remove, kept it to make sure new code was working.